### DISCUSSION TITLE
[FEATURE] Remove camera and microphone access block confirmation dialog

### DISCUSSION THREAD

**Discussion by naxitoo** [salvogiangri/UN1CA#196] [Feature requests]
### 📄 Feature description
When you press the **CameraToggleTile** or **MicrophoneToggleTile** quick setting tiles to block camera or microphone access, if enabled, a **SystemUIDialog** dialog shows up asking for confirmation. This behavior can be annoying, as it requires pulling down the Quick Settings panel again for each action.

This suggestion aims to **remove this unnecessary confirmation dialog** and **block access directly when the tiles are tapped**, without bypassing the phone unlock requirement if done from the lock screen, as expected.

https://github.com/user-attachments/assets/7223a74d-3cbd-432a-9126-6820efa66360

https://github.com/user-attachments/assets/63d35e3e-7f7a-40e4-9aee-dcab2d675a5c

<p align="center">
<img src="https://github.com/user-attachments/assets/fe5b310b-ea6a-4897-8528-e42120293d25" alt="Camera access blocking confirmation dialog" width="30%" height="auto">
&nbsp;&nbsp;<img src="https://github.com/user-attachments/assets/eb12a2af-ef2f-435a-8a4d-22ed9802439f" alt="Microphone access blocking confirmation dialog" width="30%" height="auto">
&nbsp;&nbsp;<img src="https://github.com/user-attachments/assets/a276b01e-9de1-41b0-8ff9-6e4cf248731c" alt="CameraToggleTile and MicrophoneToggleTile quick setting tiles" width="30%" height="auto">
</p>

 ### 🚀 Motivation
Although it's a small change, it saves time. Users running custom ROMs don't need to be asked for confirmation for these kinds of actions, so it will definitely enhance the user experience.

~~Additionally, there could be the possibility to make this improvement toggleable in the UN1CA settings, for those who might prefer keeping the confirmation dialog.~~

**Comment by nicodotgit** [comment]
+1 that would be a very useful implementation and would make using these functions much easier 💪
